### ISSUE TITLE
1.8.5: Unable to use private network after switching from ifconfig to ip

### ISSUE THREAD

**Issue by mvanbaak** [mitchellh/vagrant#7668]
In https://github.com/mitchellh/vagrant/pull/7460 all RedHat/CentOS guests were switched from `ipconfig` to `ip` for the network config.
Since that commit Amazon linux guests cannot be used as the networking of them is broken.

One thing I noticed while debugging is that when a guest has more then one network interface (which is the case when you specify a private network in your Vagrantfile) is that ifconfig and ip return the available nic's in different order:

```bash
vagrant@insided-dev:~$ ifconfig -a | grep -o ^[0-9a-z]* | grep -v '^lo'
eth0
eth1
vagrant@insided-dev:~$ ip -o -0 addr | grep -v LOOPBACK | awk '{print $2}' | sed 's/://'
eth1
eth0
vagrant@insided-dev:~$
```

This bug is preventing our company to use vagrant > 1.8.4, and thus preventing us from using Virtualbox > 5.0

### Vagrant version
Vagrant 1.8.5

### Host operating system
Mac OS X 10.11

### Guest operating system
Amazon linux 2016.3

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|
  # This is here because of another bug which is fixed in master
  config.ssh.insert_key = false
  config.vm.box = "mvbcoding/awslinux"
  config.vm.network "private_network", ip: "192.168.33.10"
end
```

### Debug output
```bash
==> default: Configuring and enabling network interfaces...
 INFO ssh: SSH is ready!
 INFO ssh: Execute:  (sudo=false)
 INFO guest: Autodetecting host type for [#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>]
 INFO ssh: Execute: grep 'ostree=' /proc/cmdline (sudo=false)
 INFO ssh: Execute: cat /etc/issue | grep 'Linux Mint' (sudo=false)
 INFO ssh: Execute: [ -x /usr/bin/lsb_release ] && /usr/bin/lsb_release -i 2>/dev/null | grep Trisquel (sudo=false)
 INFO ssh: Execute: grep 'Amazon Linux AMI' /etc/os-release (sudo=false)
 INFO guest: Detected: amazon!
 INFO guest: Execute capability: configure_networks [#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>, [{:type=>:static, :adapter_ip=>"192.168.33.1", :ip=>"192.168.33.10", :netmask=>"255.255.255.0", :auto_config=>true, :interface=>1}]] (amazon)
 INFO ssh: SSH is ready!
 INFO ssh: Execute:  (sudo=false)
 INFO guest: Execute capability: network_scripts_dir [#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>] (amazon)
 INFO ssh: SSH is ready!
 INFO ssh: Execute:  (sudo=false)
 INFO guest: Execute capability: network_interfaces [#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>] (amazon)
 INFO ssh: Execute: /sbin/ip -o -0 addr | grep -v LOOPBACK | awk '{print $2}' | sed 's/://' (sudo=true)
 INFO ssh: Execute: # Down the interface before munging the config file. This might
# fail if the interface is not actually set up yet so ignore
# errors.
/sbin/ifdown 'eth0' || true

# Move new config into place
mv '/tmp/vagrant-network-entry-eth0-1470128445-0' '/etc/sysconfig/network-scripts/ifcfg-eth0'

# Bring the interface up
ARPCHECK=no /sbin/ifup 'eth0'
 (sudo=true)
```

full log can be found here:
https://gist.github.com/mvanbaak/dfb5ca063456727485ce8803103d44cd
### Expected behavior
Bring up the box, have eth0 configured on the NAT, have eth1 configured for the private network

### Actual behavior
Bring up the box, try to configure eth0 for the private network, breaking networking of the guest completely

### Steps to reproduce
1. Use the provided Vagrantfile
2. run `vagrant up`

### References
- https://github.com/mitchellh/vagrant/pull/7460

**Comment by honzas** [comment]
Perhaps similar to #7613 ?

**Comment by mvanbaak** [comment]
Not really, as there the `ip` command seems to return datetime and uname output.
Looks like a different issue.

Having a quick look at the code, I think the best way to fix this one is to give plugins/guests/amazon it's own configure_network cap where it uses `ifconfig` instead of `ip`.
But I have never looked at the vagrant source before, and I'm not really sure how to do it now.

**Comment by bjoernhaeuser** [comment]
Having a similar problem with centos 7.2. The interfaces do not connect. I had to open the vbox gui and manually connect the cables. Is this the same bug or a different one?

**Comment by mvanbaak** [comment]
@bjoernhaeuser That looks like a different issue.

**Comment by agusti-t** [comment]
I just asked a question in stackoverflow since I am having a similar problem with centos/7. @mvanbaak would you say it is a similar problem?

https://stackoverflow.com/questions/38905448/vagrant-vm-doesnt-get-assigned-ip

**Comment by chrisroberts** [comment]
Fixed via #7705. Thanks for the report!
